File: apps/starter-ios/StarterApp/Core/Media/MediaModels.swift
import Foundation

struct PrepareMediaUploadBody: Encodable {
    let kind: String
    let mimeType: String
    let fileName: String?
    let sizeBytes: Int?
    let visibility: String?
}

struct CompleteMediaUploadBody: Encodable {
    let assetId: String
    let width: Int?
    let height: Int?
}

struct MediaUploadPreparation: Codable, Equatable {
    let assetId: String
    let storageKey: String
    let uploadUrl: String
    let publicUrl: String?
    let headers: [String: String]
    let expiresInSeconds: Int
}

struct MediaAsset: Codable, Equatable {
    let id: String
    let ownerId: String
    let kind: String
    let status: String
    let storageKey: String
    let publicUrl: String?
    let mimeType: String
    let fileName: String?
    let sizeBytes: Int?
    let width: Int?
    let height: Int?
    let visibility: String
    let createdAt: String
}

File: apps/starter-android/app/src/main/kotlin/com/onlystack/starterapp/core/media/MediaModels.kt
package com.onlystack.starterapp.core.media

data class PrepareMediaUploadBody(
    val kind: String,
    val mimeType: String,
    val fileName: String? = null,
    val sizeBytes: Int? = null,
    val visibility: String? = null,
)

data class CompleteMediaUploadBody(
    val assetId: String,
    val width: Int? = null,
    val height: Int? = null,
)

data class MediaUploadPreparation(
    val assetId: String,
    val storageKey: String,
    val uploadUrl: String,
    val publicUrl: String? = null,
    val headers: Map<String, String>,
    val expiresInSeconds: Int,
)

data class MediaAsset(
    val id: String,
    val ownerId: String,
    val kind: String,
    val status: String,
    val storageKey: String,
    val publicUrl: String? = null,
    val mimeType: String,
    val fileName: String? = null,
    val sizeBytes: Int? = null,
    val width: Int? = null,
    val height: Int? = null,
    val visibility: String,
    val createdAt: String,
)
